docs/api/memory/link_time_optimization.md

```diff
@@ -0,0 +1,66 @@
+# Link Time Optimization
+
+Link Time Optimization (LTO) is a program memory usage optimization mechanism that the compiler performs at link time. At compile time, the compiler creates a special intermediate representation of all translation units. It then optimizes them as a single unit at link time, which uses less memory than non-LTO builds.
+
+## Using LTO in Mbed OS
+
+The Mbed OS build system implements LTO as an optional profile extension in `tools\profiles\extensions\lto.json`. Enabling LTO amends the build profile with LTO flags. LTO performs heavy memory optimizations that break debugging, so we don't recommend enabling it in debug or develop profiles.
+
+To enable LTO, retype the `--profile` option with LTO file path `tools\profiles\extensions\lto.json`.
+
+<span class="notes">**Note**: For profile extensions you have to put the full path relative to the project's root folder.</span>
+
+To enable LTO with the `release` profile:
+
+```
+mbed compile -t TOOLCHAIN -m TARGET --profile release --profile mbed-os/tools/profiles/extensions/lto.json
+```
+
+Example LTO profile memory savings for [mbed-os-example-blinky](https://github.com/ARMmbed/mbed-os-example-blinky):
+
+|Build type|Total static RAM memory (data + BSS)|Total flash memory (text + data)|
```

Comment:
What's BSS?

Reply:
All `data` `BSS` and `text` are a segments of memory to store different type of variables/data